tias/src/Components/Scheduling/SchedulingColumn.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { FC } from 'react';
+import { SchedulingBlock } from './SchedulingBlock';
+
+const numHours = 11;
+interface Props {
+  blocks: any,
+  end? : boolean
+}
+
+interface CourseInstance { // Results of a join between course, course_section, and section_meetings
+  course: number,   // Course_Number from Course
+  section: number,  // Section_Number from Course_Section
+  start: Date,      // Start_Time from Section_Meeting
+  end: Date         // End_Time from Section_Meeting 
+  // If the API returns more information from this I can add them to the interface here
+}
+
+// Breaks for 3 different lengths in a block
+const d_len = (e: CourseInstance | {course: number, section: number, start: Date, end: Date}) => { return e.end.getTime() - e.start.getTime(); }
+
+const time_to_height = (start: Date, end: Date, max: number = numHours*60*60*1000) => {
+  return (end.getTime() - start.getTime()) / max * 100;
+}
+
+const start_time_to_top = (e: any, start: Date, pstart: Date = new Date(0), parent: number = numHours*60*60*1000) => {
```

Comment:
Careful about potential overflow because all numbers in TS/JS are of type `number` which is equivalent to `float_64t`. As such, you can have big problems with precision for large values.

Reply:
The precision doesn't need to be all that precise for placing elements on the page, but if stuff starts moving around I'll keep this in mind